pycbc/io/live.py

```diff
@@ -477,39 +497,21 @@ def upload(self, fname, gracedb_server=None, testing=True,
                                gid)
                 logging.error(str(exc))
 
-        # Upload em_bright properties JSON
-        if self.hasmassgap is not None:
-            try:
-                self.gracedb.write_log(
-                    gid, 'EM Bright properties JSON file upload',
-                    filename=self.embright_file,
-                    tag_name=['em_bright']
-                )
-                logging.info('Uploaded em_bright properties for %s', gid)
-            except Exception as exc:
-                logging.error('Failed to upload em_bright properties file '
-                              'for %s', gid)
-                logging.error(str(exc))
-
-        # Upload multi-cpt p_astro JSON
-        if self.astro_probs is not None:
-            try:
-                self.gracedb.write_log(
-                    gid, 'Multi-component p_astro JSON file upload',
-                    filename=self.multipa_file,
-                    tag_name=['p_astro'],
-                    label='PASTRO_READY'
-                )
-                logging.info('Uploaded multi p_astro for %s', gid)
-            except Exception as exc:
-                logging.error('Failed to upload multi p_astro file for %s', gid)
-                logging.error(str(exc))
-            # Don't do anything else!
-            return gid
-
-        # If there is no multi p_astro, upload source probabilities in JSON
-        # format and plot
-        if self.probabilities is not None:
+        # If 'self.prob_file' exists, make and upload source probabilities
+        # in JSON format and pie plot
+        if hasattr(self, 'prob_file'):
+            self.prob_plotf = self.prob_file.replace('.json', '.png')
+            # Don't try to plot zero probabilities
+            prob_plot = {k: v for (k, v) in self.probabilities.items()
+                         if v != 0.0}
+            labels, sizes = zip(*prob_plot.items())
+            colors = [source_color(label) for label in labels]
+            fig, ax = pl.subplots()
+            ax.pie(sizes, labels=labels, colors=colors, autopct='%1.1f%%',
+                   textprops={'fontsize': 15})
+            ax.axis('equal')
+            fig.savefig(self.prob_plotf)
+            pl.close()
```

Comment:
So it seems the pie chart would still only show the astrophysical classes right? I.e. if a trigger has a total p_astro of 0.5 divided equally between BNS and NSBH, would the pie chart show 0.5 BNS and 0.5 NSBH? This may be ok provided that it is extremely clear that the plot is only showing the astrophysical classes.

Reply:
Personally I would plot *every* class including terrestrial, if available and if nonzero, but maybe this can be a later PR if agreed.

Reply:
This code to make an 'astrophysical class' pie chart has always done only that, ie it ignores p_terr, the choice of a pie format was made specifically to differentiate it from the standard horizontal bar multi component p astro plot.

Reply:
(I don't think we want a pie that just duplicates the content of the gwcelery style bar plot)

Reply:
Fair enough, though I might request adding a comment to explicitly state the intention (for when I will forget this).

Reply:
Also, I would not move this block from where it was - it should still happen before the "Don't try to do anything else!" check.

Reply:
Well, you asked for this plot to come after the SNR time series and ASD uploads. So then the solution would be to move or remove the `if gid is None` block

Reply:
I guess the cleanest way to handle this would be remove that block, but put a test for `gid is not None` in the final annotation.